core/includes/file.inc

```diff
@@ -2018,6 +2018,66 @@ function file_validate_image_orientation(File $file, $orientation = 0) {
   return $errors;
 }
 
+/**
+ * Validate uploaded SVG files.
+ *
+ * @param File $file
+ *   A file entity.
+ *
+ * @return
+ *   An array. If the file is an SVG and is either not valid or contains
+ *   dangerous content, it will contain an error message.
+ *
+ * @see hook_file_validate()
+ */
+function file_validate_svg(File $file) {
+  $errors = array();
+
+  if ($file->filemime == 'image/svg+xml') {
+    $file_contents = file_get_contents($file->uri);
+    $xml = @simplexml_load_string($file_contents);
```

Comment:
Suppressing errors with `@` here contradicts the way we try to evaluate errors in the next condition.

I'd still strongly recommend to wrap our libxml checks inside libxml_use_internal_errors() to actually *capture* them.
I can provide a sample file, where this makes a real difference.

Reply:
The `@` just suppresses *displaying* the errors, just to hide all PHP error messages from displaying (if the file is invalid, PHP throws a *whole bunch* of messages about xml elements. we dont need those here, theyre in the log) . All we need for the upload validator is the simple message. Using `libxml_use_internal_errors()` doesnt help with this AFAICT.

Reply:
> we dont need those here, theyre in the log...

TBH, I wouldn't want a crappy file to clutter the log, when uploading.

> Using libxml_use_internal_errors() doesnt help with this AFAICT.

It does in my local testing. We could even process them to display a more verbose message, to tell people why this file is crap.
This error handling might eventually be different in different PHP versions. Will do further testing later.

Reply:
A lot more testing... When using libxml_use_internal_errors() (true/false) wrapped around, *nothing* shows up in the logs.

Tested with PHP 5.6.40, 7.4.33 and 8.2.18 in combination with libxml version 2.9.14.
Can we please use libxml_use_internal_errors? It's more useful here than just the `@` operator.

Reply:
> If you want to take over this one it would be great, I am also working on other PRs too...

I belief, this PR is in a good shape and already a valid candidate for 1.28. Remember: it doesn't have to be fully done by feature freeze. Just ready enough to look promising - and I think it is. :wink: 

Yes, I'm aware that you're involved in other 1.28 candidates. That's a bit unfortunate, and I hope, Backdrop doesn't eat too much of your sparse spare time. :pleading_face:

I don't think, it's necessary that I take over. I'll just try to be more patient. :innocent: Didn't mean to stress you out.

Reply:
:smiley: phew!
Will think it over.

Reply:
If there are just small changes here and there I could edit @docwilmot's branch.